confmap/provider.go

```diff
@@ -101,17 +102,39 @@ func WithRetrievedClose(closeFunc CloseFunc) RetrievedOption {
 }
 
 // NewRetrieved returns a new Retrieved instance that contains the data from the raw deserialized config.
-func NewRetrieved(rawConf map[string]interface{}, opts ...RetrievedOption) (Retrieved, error) {
+// The rawConf can be one of the following types:
+//   * Primitives: int, int32, int64, float32, float64, bool, string;
+//   * []interface{};
+//   * map[string]interface{};
+func NewRetrieved(rawConf interface{}, opts ...RetrievedOption) (Retrieved, error) {
+	if err := checkRawConfType(rawConf); err != nil {
```

Comment:
I think it would be good to have a unit test passing a `rawConf` of an unsupported type

Reply:
Done.